### PR TITLE
Add migration guide for `RawKeyEvent` to `KeyEvent` transition

### PR DESCRIPTION
**Pull request by gspencergoog**
## Description

Adds a migration guide for the `RawKeyEvent`/`RawKeyboard` migration.

## Related Issues
 - https://github.com/flutter/flutter/issues/136419

### CODE REVIEW

File: src/release/breaking-changes/index.md [hunk 1]
```diff
@@ -39,6 +39,7 @@ release, and listed in alphabetical order:
 
 ### Released in Flutter 3.16
 
+* [Migrate RawKeyEvent/RawKeyboard system to KeyEvent/HardwareKeyboard system][]
```
Comment (Piinks) [inline src/release/breaking-changes/index.md:42]: Should this be under 'Not yet released to stable'?

File: src/release/breaking-changes/key-event-migration.md [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+
+## Summary
+
+For some time now (years), Flutter has had two key event systems implemented.
+The new system reached parity with the old platform-specific raw key event
+system, and the raw system has been deprecated.
+
+## Context
+
+In the original key event subsystem handling each platform's quirks in the
```
Comment (Piinks) [inline src/release/breaking-changes/key-event-migration.md:17]: Nit
```suggestion
In the original key event subsystem, handling each platform's quirks in the
```

**Review by Piinks — APPROVED** [review]

File: src/release/breaking-changes/index.md (hunk 1 above)
Comment (sfshaza2) [reply src/release/breaking-changes/index.md:42]: Yes, please list this under the header "Not yet released to stable".

**Review by sfshaza2 — APPROVED** [review]
lgtm, once you've put the new breaking change under the correct header in the index file. Thanks, @gspencergoog, this is a big one!

**Pull request merged**